### ISSUE TITLE
When running integration tests with xunit on netcoreapp ScriptLocation returns an invalid path

### ISSUE THREAD

**Issue by DavidBoike** [Particular/NServiceBus.Persistence.Sql#340]
Replaces #312

which means scripts dont get executed

`file:///C:/Users/simon/.nuget/packages/microsoft.testplatform.testhost/15.0.0/lib/netstandard1.5/testhost.dll`

![image](https://user-images.githubusercontent.com/122666/46011482-503f6b00-c109-11e8-9e3a-200bb1ac1caf.png)

seems `Assembly.GetEntryAssembly()` returns the testhost location. which is invalid. While the `AppDomain.CurrentDomain.BaseDirectory` is correct but is doesnt get inside the if

workaround: run them manually

```

    static Task RunSqlScripts(string endpointName, Func<DbConnection> connectionBuilder)
    {
        var baseDir = AppDomain.CurrentDomain.BaseDirectory;
        var scriptDir = Path.Combine(baseDir, "NServiceBus.Persistence.Sql", "MsSqlServer");

        return ScriptRunner.Install(
                sqlDialect: new SqlDialect.MsSqlServer(),
                tablePrefix: endpointName+"_",
                connectionBuilder: connectionBuilder,
                scriptDirectory: scriptDir,
                shouldInstallOutbox: false,
                shouldInstallSagas: true,
                shouldInstallSubscriptions: false,
                shouldInstallTimeouts: false);
    }
```

**Comment by mikesigs** [comment]
I tried the work-around above, but it doesn't really resolve my issue. It does in fact run through the scripts, but it still fails later in the test.

Our integration test is starting the endpoint, without any test-specific modifications to the EndpointConfiguration. Our endpoint is configured with `.EnableInstallers()`. So when it hits the `RunInstallers` step during startup it tries to run the SQL scripts. It fails with a very similar error message to this Issue:

`Expected 'C:\Users\mikes\.nuget\packages\microsoft.testplatform.testhost\15.9.0\lib\netstandard1.5\NServiceBus.Persistence.Sql\MsSqlServer\Outbox_Create.sql' to exist. It is possible it was not deployed with the endpoint.`

I'll keep digging through this on my own, but was hoping there might be some progress on permanent fix for this issue.

Thanks!

**Comment by DavidBoike** [comment]
@mikesigs we're getting ready to release the fix to this. Stay tuned.

**Comment by InLoop-TOCA** [comment]
Hi @DavidBoike 

Has this issue been fixed? If yes then in which version of NServiceBus.Persistence.Sql ?

Thanks

**Comment by DavidBoike** [comment]
@InLoop-TOCA Yes, it was fixed in 4.6.0. Sorry, the issue #312 that this one replaced is the one that got used for the release notes, and so that one is tied to the milestone showing the version.
